salt/states/file.py

```diff
@@ -676,7 +676,7 @@ def _check_directory(name,
                      user=None,
                      group=None,
                      recurse=False,
-                     mode=None,
+                     dir_mode=None,
```

Comment:
I am not entirely sure if no other parts call this function by kwargs, it could break things

Reply:
It looks like calls to this is contained within `states/file.py` and the calls here all use positional arguments:
```
 $ grep -r _check_directory .
./states/file.py:def _check_directory(name,
./states/file.py:def _check_directory_win(name,
./states/file.py:        presult, pcomment, pchanges = _check_directory_win(
./states/file.py:        presult, pcomment, pchanges = _check_directory(
./states/file.py:                check_res, check_msg, ret['pchanges'] = _check_directory_win(dirname)
./states/file.py:                check_res, check_msg, ret['pchanges'] = _check_directory(dirname)
./states/file.py:                check_res, check_msg, ret['pchanges'] = _check_directory_win(dirname)
./states/file.py:                check_res, check_msg, ret['pchanges'] = _check_directory(dirname)
```